http/http_rest_proxy.h

```diff
@@ -258,36 +256,28 @@ struct HttpRestProxy {
     /** Are we debugging? */
     bool debug;
 
-private:    
-    /** Lock for connection pool. */
-    mutable std::mutex lock;
-
-    /** List of inactive handles.  These can be selected from when a new
-        connection needs to be made.
-    */
-    //TODO replace with a safer storage - typedef std::shared_ptr<CurlWrapper::Easy> EasyPtr;
-    mutable std::vector<CurlWrapper::Easy*> inactive;
-
-    std::vector<std::string> cookies;
+private:
+    /** Private type that implements a connection handler. */
+    struct ConnectionHandler;
 
 public:
     /** Get a connection. */
     struct Connection {
-        Connection(CurlWrapper::Easy * conn,
+        Connection(ConnectionHandler * conn,
                    HttpRestProxy * proxy)
             : conn(conn), proxy(proxy)
         {
         }
 
-        ~Connection();
+        ~Connection() noexcept;
```

Comment:
@FinchPowers  a raison, pour une certaine définition de "par défaut": (from http://en.cppreference.com/w/cpp/language/noexcept_spec):

"Inheriting constructors and the implicitly-declared default constructors, copy constructors, move constructors, destructors, copy-assignment operators, move-assignment operators are all noexcept(true) by default, unless they are required to call a function that is noexcept(false), in which case these functions are noexcept(false). "
+
"Any user-defined destructor is noexcept(true) by default, unless the declaration specifies otherwise, or the destructor of any base or member is noexcept(false). "